include/autoppl/expression/distribution/discrete.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <cassert>
+#include <random>
+#include <cmath>
+#include <numeric>
+
+namespace ppl {
+
+// TODO: change name to DiscreteDist and make class template.
+// Discrete should be a function that creates this kind of object.
+
+template <typename weight_type>
+struct Discrete 
+{
+    using value_t = int;
```

Comment:
can you do `uint64_t` since this will always output from `[0,n)`. 64 should be more than enough for large values xD

Reply:
addressed all your comments in a new commit